src/dashboard/src/components/filesystem_ajax/views.py

```diff
@@ -628,87 +588,103 @@ def copy_to_arrange(request, sourcepath=None, destination=None):
+                original_path=entry['original_path'],
+                arrange_path=entry['arrange_path'],
+                file_uuid=entry['file_uuid'],
+                transfer_uuid=entry['transfer_uuid'],
+            )
+        except IntegrityError:
+            # FIXME Expecting this to catch duplicate original_paths, which
+            # we want to ignore since a file can only be in one SIP.  Needs
+            # to be updated not to ignore other classes of IntegrityErrors.
+            logging.exception('Integrity error inserting: %s', entry)
 
-        logging.info('copy_to_arrange: arrange_path: {}'.format(arrange_path))
-        logging.debug('copy_to_arrange: files to be added: {}'.format(to_add))
 
-        for entry in to_add:
-            try:
-                # TODO enforce uniqueness on arrange panel?
-                models.SIPArrange.objects.create(
-                    original_path=entry['original_path'],
-                    arrange_path=entry['arrange_path'],
-                    file_uuid=entry['file_uuid'],
-                    transfer_uuid=entry['transfer_uuid'],
-                )
-            except IntegrityError:
-                # FIXME Expecting this to catch duplicate original_paths, which
-                # we want to ignore since a file can only be in one SIP.  Needs
-                # to be updated not to ignore other classes of IntegrityErrors.
-                logging.exception('Integrity error inserting: %s', entry)
+def copy_to_arrange(request, sourcepath=None, destination=None):
+    """ Add files to in-progress SIPs being arranged.
+    Files being copied can be located in either the backlog or in another SIP being arranged.
 
-    if error is not None:
+    sourcepath: GET parameter, path relative to this pipelines backlog. Leading
+        '/'s are stripped
+    destination: GET parameter, path within arrange folder, should start with
+        DEFAULT_ARRANGE_PATH ('/arrange/')
+    """
+    # Insert each file into the DB
+
+    if sourcepath is None:
+        sourcepath = base64.b64decode(request.POST.get('filepath', ''))
+    if destination is None:
+        destination = base64.b64decode(request.POST.get('destination', ''))
+    logging.info('copy_to_arrange: sourcepath: {}'.format(sourcepath))
+    logging.info('copy_to_arrange: destination: {}'.format(destination))
+
+    try:
+        if sourcepath.startswith('/' + DEFAULT_BACKLOG_PATH):
+            copy_files_to_arrange(sourcepath, destination)
+            response = {'message': 'Files added to the SIP.'}
+            status_code = 201
+        elif sourcepath.startswith(DEFAULT_ARRANGE_PATH):
+            move_files_within_arrange(sourcepath, destination)
+            response = {'message': 'SIP files successfully moved.'}
+            status_code = 200
+        else:
+            raise ValueError('filepath is in neither the base backlog path nor the arrange path ({} provided)'.format(sourcepath))
+    except ValueError as e:
```

Comment:
Log the exception as well?

Also, we've figured out more about logging since this was written and it should be `logger.foo` instead of `logging.foo` for all logging calls so it doesn't go to the root logger.